File: code/python/command/Waiter.py
class Waiter:
    def __init__(self):
        self.command = None

    def set_command(self, command):
        self.command = command

    def execute_command(self):
        print(f"Waiter.executeCommand(): Delegating to Command.execute() for {type(self.command).__name__}")
        self.command.execute()

File: code/typescript/src/command/Waiter.ts
import { Command } from "./Command";

export class Waiter {
    private command!: Command;

    setCommand(command: Command): void {
        this.command = command;
    }

    executeCommand(): void {
        console.log(`Waiter.executeCommand(): Delegating to Command.execute() for ${this.command.constructor.name}`);
        this.command.execute();
    }
}
